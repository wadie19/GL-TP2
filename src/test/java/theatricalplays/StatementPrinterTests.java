package theatricalplays;

//import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import static org.approvaltests.Approvals.verify;
import static org.approvaltests.Approvals.verifyHtml;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class StatementPrinterTests {

    @Test
    void exampleStatement() {
        Map<String, Play> plays = Map.of(
                "hamlet",  new PlayTragedy("Hamlet"),
                "as-like", new PlayComedy("As You Like It"),
                "othello", new PlayTragedy("Othello"));

        Invoice invoice = new Invoice(new Customer("BigCo", "1", 103), List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)));

        StatementPrinter statementPrinter = new StatementPrinter();
        var result = statementPrinter.print(invoice, plays);

        verify(result);
    } 

    @Test
    void exampleStatementCustomer() {
        Map<String, Play> plays = Map.of(
                "hamlet",  new PlayTragedy("Hamlet"),
                "as-like", new PlayComedy("As You Like It"),
                "othello", new PlayTragedy("Othello"));

        Invoice invoice = new Invoice(new Customer("BigCo", "1", 150), List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)));
        
        StatementPrinter statementPrinter = new StatementPrinter();
        var result = statementPrinter.print(invoice, plays);

        verify(result);
    }

    @Test
    void exampleStatementHTML() {
        Map<String, Play> plays = Map.of(
                "hamlet",  new PlayTragedy("Hamlet"),
                "as-like", new PlayComedy("As You Like It"),
                "othello", new PlayTragedy("Othello"));

        Invoice invoice = new Invoice(new Customer("BigCo", "2", 0), List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)));

        StatementPrinter statementPrinter = new StatementPrinter();
        var result = statementPrinter.toHTML(invoice, plays);

        verifyHtml(result);
    }

    /*@Test
    void statementWithNewPlayTypes() {
        Map<String, Play> plays = Map.of(
                "henry-v",  new Play("Henry V", "history"),
                "as-like", new Play("As You Like It", "pastoral"));

        Invoice invoice = new Invoice("BigCo", List.of(
                new Performance("henry-v", 53),
                new Performance("as-like", 55)));

        StatementPrinter statementPrinter = new StatementPrinter();
        Assertions.assertThrows(Error.class, () -> {
        statementPrinter.print(invoice, plays);
        });
    }*/

    @Test
    void testAudience30ForTragedy() {
    HashMap<String, Play> plays = new HashMap<>();
    plays.put("playID", new PlayTragedy("Play"));

    Invoice invoice = new Invoice(new Customer("Customer", "1", 0), List.of(new Performance("playID", 30)));

    StatementPrinter statementPrinter = new StatementPrinter();
    var result = statementPrinter.print(invoice, plays);

    StringBuffer expectedStatement = new StringBuffer();
    expectedStatement.append("Statement for Customer\n");
    expectedStatement.append("  Play: $400.00 (30 seats)\n");
    expectedStatement.append("Amount owed is $400.00\n");
    expectedStatement.append("You earned 0 credits\n");

    assertEquals(expectedStatement.toString(), result.toString());
}


    @Test
    void testAudience20ForComedy() {
    HashMap<String, Play> plays = new HashMap<>();
    plays.put("playID", new PlayComedy("Play"));

    Invoice invoice = new Invoice(new Customer("Customer", "1", 0), List.of(new Performance("playID", 20)));

    StatementPrinter statementPrinter = new StatementPrinter();
    var result = statementPrinter.print(invoice, plays);

    StringBuffer expectedStatement = new StringBuffer();
    expectedStatement.append("Statement for Customer\n");
    expectedStatement.append("  Play: $360.00 (20 seats)\n");
    expectedStatement.append("Amount owed is $360.00\n");
    expectedStatement.append("You earned 4 credits\n");

    assertEquals(expectedStatement.toString(), result.toString());
    }
}
