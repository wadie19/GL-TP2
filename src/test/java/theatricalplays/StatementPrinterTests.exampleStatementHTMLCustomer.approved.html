<!DOCTYPE html> 
<html> 
	<head>
		<style>
			table, th, td {border: 1px solid black; }
			th, td { padding: 5px; }
		</style>
		</head>
	<body> 
		<h1>Invoice</h1> 
		<ul><li><p><b>Client :</b> BigCo</ul></li></h2> 
		<table style="width: 400px; border: 1px solid black"> 
			<tr> 
				 <th>Piece</th> 
				 <th>Seats sold</th> 
				 <th>Price</th> 
			</tr> 
			<tr> 
				 <td>Hamlet</td> 
				 <td>55</td> 
				 <td>$650.00</td> 
			</tr> 
			<tr> 
				 <td>As You Like It</td> 
				 <td>35</td> 
				 <td>$580.00</td> 
			</tr> 
			<tr> 
				 <td>Othello</td> 
				 <td>40</td> 
				 <td>$500.00</td> 
			</tr> 
				<tr> 
					 <td colspan="2" style="text-align: right;"><b>Total owed:<b></td> 
					 <td>$1,715.00</td> 
				</tr> 
				<tr> 
					 <td colspan="2" style="text-align: right;"><b>Fidelity points earned:<b></td> 
					 <td>0</td> 
				</tr> 
			</table> 
			<p><i>Congrats! You have been deducted of 150 credits and $15.00<i></p> 
			<p><i>Payment is required under 30 days. We can break your knees if you don't do so.<i></p> 
		</body> 
	</html> 
